RIPS/rip-9999.md

```diff
@@ -0,0 +1,313 @@
+
+None of these proposals supports Gas Abstraction mechanism that would be conceptually similar to "Paymasters" - smart
+contracts capable of paying transaction gas, based on some on-chain validity function, but do not gain any
+privileged access to sender's funds.
+This is very disappointing given that the concept of "Paymasters" has being widely used in production by
+[ERC-4337: Account Abstraction Using Alt Mempool](https://eips.ethereum.org/EIPS/eip-4337)
+and even
+[ERC-1613: Gas Stations Network](https://eips.ethereum.org/EIPS/eip-1613)
+before that.
+
+None of these proposals fully support validity time ranges, which enable both scheduled and expiring transactions,
+adding a noticeable improvement to end-user experience with a relatively small amount of effort.
+
+And most importantly, none of these proposals form a part of a bigger vision.
+Instead, these proposals are isolated patches that can be applied in a single design iteration.
+These proposals cannibalize each other's use-cases, but do not fully resolve the pain points Ethereum users are
+currently suffering from, and they do not fully align with any kind of vision for a Full Native Account Abstraction.
+
+## Specification
+The aforementioned EIPs can be implemented by the following transaction subtypes:
+
+### Shared RIP-9999 transaction payload
+
+The required synergy of the proposed transaction subtypes is achieved by defining a shared base structure of
+a transaction.
+This will ensure the only differences between these subtypes are the ones strictly required
+by their semantics instead of being caused by different tastes or variations in implementation details.
+
+These are the shared parameters of any RIP-9999 transaction subtype payload:
+
+```
+  chainId,
+  nonce,
+  callData,
+  paymasterData,
+  paymasterGasLimit,
+  maxPriorityFeePerGas,
+  maxFeePerGas,
+  validAfter,
+  validUntil,
+  accessList
+```
+
+### Gas-sponsored time-limited EOA transaction subtype
+
+```
+SPONSORED_EOA_SUBTYPE = x
+```
+
+This RIP-7560 transaction subtype allows ECDSA signature and nonce validation, same as the one used in all
+of existing transaction types, to be applied to a transaction that can also provide `paymasterData`, `validAfter`
+and `validUntil` parameters.
+
+The transaction payload for this subtype is extended with the following fields:
+```
+ to
+ value
+ callGasLimit,
+ yParity, r, s
+```
+
+This subtype extends the ability of an RIP-7560 transaction to specify a Paymaster contract, 
+which accepts the gas charges, to all EOA accounts.
+It also provided an EOA with an ability to explicitly specify a time range where this
+transaction is valid, creating both "expiring" and "delayed" transactions.
+
+In case the `to` field is `nil`, `callData` is treated as `initCode` for the purpose of contract creation,
+enabling Gas Abstracted new smart contract deployment.
+
+
+
+### Gas-sponsored time-limited batched execution EOA transaction subtype
+
+
+```
+BATCHED_EOA_SUBTYPE = x
```

Comment:
There is some redundancy in these transaction types of course. Delegated call can be functionally equivalent to a batched call, which is equivalent to a sponsored call for a batch with a single transaction.

I think one issue is a question of a slightly increased 'trust surface'. Delegating control of an EOA to a smart contract means fully trusting this contract, and signing such a transaction will need a new defence level from the wallet apps. Validating a `Multicall` code is simply not something they currently do. Signing a batch, on the other hand, is exactly what they do now - show the payload and ask for confirmation.

Another is a question of EOA's storage. It is not currently defined and there are arguments not to introduce it yet. In this case the EOA's context will be different from other EVM `staticcall` contexts because it cannot use SSTORE itself but can call other contract's code that uses SSTORE.

We can discuss which of the subtypes make most sense and remove the redundant ones.